services/pushover.py

```diff
@@ -77,9 +95,37 @@ def plugin(srv, item):
     if callback is not None:
         params['callback'] = callback
 
+    srv.logging.debug("=================Starting Image processing")
+
+    filepayload=None
+    try:
+        srv.logging.debug("Loading mesasgejson")
+
+        messagejson = json.loads(message)
+        srv.logging.debug("Mesasgejson loaded, looking for imageurl")
+
+        if 'imageurl' in messagejson:
+            srv.logging.debug("Image url is in json object %s" % messagejson["imageurl"])
+            filepayload = { "attachment": ("image.jpg",requests.get(messagejson["imageurl"], stream=True).raw, "image/jpeg") }
+            message = messagejson["message"]
+        elif 'image' in messagejson:
+            srv.logging.debug("Image is in json object %s" % messagejson["image"])
+            filepayload = { "attachment": ("image.jpg",base64.decodestring(messagejson["image"]), "image/jpeg") }
+            message = messagejson["message"]
+        else:
+            srv.logging.debug("No image could be found")
+
+
+
+    except Exception, e:
+        srv.logging.warn("Error sending pushover notification to %s [%s]: %s" % (item.target, params, str(e)))
```

Comment:
This logging message is wrong - an exception here just means we didn't have a suitable JSON payload. Would be better to catch the decoding exception from `json.loads()` and `pass`, and handle any other exception with a suitable error message and fail the operation.

Reply:
Think the expected error is `JSONDecodeError`

Reply:
Yeah unsure.  The try will catch an error decoding the json, so yes the message to logging should change.   But as it is, it will also skip all the attempts at parsing that json that it couldn't read.

I think I'd be better to update the error to "could not parse json etc", and then fail out of the routine.

So like this:
```
except JSONDecodeError, jsone:
    srv.logging.error("Error decoding json message when processing images %s [%s]: %s" % (item.target, params, str(jsone)))
    return False
except Exception, e:
    srv.logging.error("Error parsing json parameters in sending pushover images to %s [%s]: %s" % (item.target, params, str(e)))
    return False
```

Reply:
So the only bit I'm unsure of is the final else:
```
else:
            srv.logging.debug("No image could be found,")
```
basically this will send out a json payload to pushover, as it's decoded it, but it didn't contain either of the two image formats.  I'm kind happy with that, should only happen if you screw up....unless I just pull the "message" attribute out and just send that as a text message only